spec.html

```diff
@@ -30384,6 +30407,50 @@ <h1>Runtime Semantics: GetSubstitution ( _matched_, _str_, _position_, _captures
         </emu-clause>
       </emu-clause>
 
+      <emu-clause id="sec-string.prototype.replaceall">
+        <h1>String.prototype.replaceAll ( _searchValue_, _replaceValue_ )</h1>
+        <p>When the `replaceAll` method is called with arguments _searchValue_ and _replaceValue_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _O_ be ? RequireObjectCoercible(*this* value).
+          1. If _searchValue_ is neither *undefined* nor *null*, then
+            1. Let _isRegExp_ be ? IsRegExp(_searchValue_).
+            1. If _isRegExp_ is *true*, then
+              1. Let _flags_ be ? Get(_searchValue_, *"flags"*).
+              1. Perform ? RequireObjectCoercible(_flags_).
+              1. If ? ToString(_flags_) does not contain *"g"*, throw a *TypeError* exception.
+            1. Let _replacer_ be ? GetMethod(_searchValue_, @@replace).
+            1. If _replacer_ is not *undefined*, then
+              1. Return ? Call(_replacer_, _searchValue_, « _O_, _replaceValue_ »).
+          1. Let _string_ be ? ToString(_O_).
+          1. Let _searchString_ be ? ToString(_searchValue_).
+          1. Let _functionalReplace_ be IsCallable(_replaceValue_).
+          1. If _functionalReplace_ is *false*, then
+            1. Set _replaceValue_ to ? ToString(_replaceValue_).
+          1. Let _searchLength_ be the length of _searchString_.
+          1. Let _advanceBy_ be max(1, _searchLength_).
+          1. Let _matchPositions_ be a new empty List.
+          1. Let _position_ be ! StringIndexOf(_string_, _searchString_, 0).
+          1. Repeat, while _position_ is not -1,
```

Comment:
Your test in String.prototype.includes is `If _index_ &ge; 0`. I don't have a preference for "≥ 0" vs. "is not -1", but I do have a preference for them to match.

Reply:
I've made the change. This is a little awkward though:

```
If index is not -1, return true.
Return false.
```

In code you'd probably write it as:

```
If index is -1, return false.
Return true.
```

🤷